s3_storage_provider.py

```diff
@@ -60,7 +60,7 @@ class S3StorageProviderBackend(StorageProvider):
     """
 
     def __init__(self, hs, config):
-        self.cache_directory = hs.config.media_store_path
+        self.cache_directory = hs.config.media.media_store_path
```

Comment:
I think we might be able to delete this entirely, its only used in `store_file` as part of `os.path.join`, but the path `store_file` gets appears to be absolute already, so its pointless. I think.